test/web/test_server.py

```diff
@@ -69,3 +77,66 @@ def test_parse_address(source, result):
 
     else:
         assert parse_address(source) == result
+
+
+def test_custom_wsgi_app_is_used(container_factory, web_config):
+    wsgi_app = Mock()
+
+    class CustomWebServer(WebServer):
+        def get_wsgi_app(self):
+            return wsgi_app
```

Comment:
Is there a meaningful example we could demonstrate here, rather than using a mock? Something like applying a fixer, without too adding too much code.